nemo/collections/tts/inference/text_processors.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List
+
+from nemo_text_processing.text_normalization.normalize import Normalizer
+
+from nemo.collections.common.tokenizers.text_to_speech.tts_tokenizers import IPATokenizer
+from nemo.collections.tts.inference.inference_modules import GraphemeToPhoneme, TextProcessor, TextTokenizer
+
+
+class BaseTextProcessor(TextProcessor):
+    def __init__(self, normalizer: Normalizer):
+        self.normalizer = normalizer
+
+    def preprocess_text(self, text: str) -> str:
+        return text
+
+    def normalize_text(self, text: str) -> str:
+        normalized_text = self.normalizer.normalize(text, punct_pre_process=True, punct_post_process=True)
```

Comment:
IIRC, our normalization function call doesn't support multi-processing. So we may need to enforce a multi-processing in this function for better computing efficiency.

Reply:
FYI, i used this before: https://github.com/NVIDIA/NeMo/blob/main/scripts/dataset_processing/tts/hui_acg/get_data.py#L201-L202

Reply:
a second look, it seems your design only takes as input a single `str` while not a batch. Maybe the multiprocessing should be moved to `pipeline.py`.